src/com/facebook/buck/ide/intellij/IjSourceRootSimplifier.java

```diff
@@ -330,6 +340,60 @@ private FolderTypeWithPackageInfo findBestFolderType(ImmutableCollection<IjFolde
       }
     }
 
+    /** Merges JavaResourceFolders or JavaTestResourceFolders. */
+    private Optional<IjFolder> tryCreateNewParentFolderFromChildrenResourceFolders(
+        Path currentPath,
+        ImmutableCollection<IjFolder> children,
+        FolderTypeWithPackageInfo typeForMerging) {
+      IjResourceFolderType ijResourceFolderType = typeForMerging.getIjResourceFolderType();
+
+      ImmutableList<IjFolder> childrenToMerge =
+          children
+              .stream()
+              .filter(ijResourceFolderType::isIjFolderInstance)
+              .collect(MoreCollectors.toImmutableList());
+
+      if (childrenToMerge.isEmpty()) {
+        return Optional.empty();
+      }
+
+      final Path resourcesRoot =
+          ijResourceFolderType.getResourcesRootFromFolder(childrenToMerge.get(0));
+
+      // If merging would recurse above resources_root, don't merge.
+      if (!currentPath.startsWith(resourcesRoot)) {
```

Comment:
It looks like `resourcesRoot` can be `null`, which would make this throw a `NullPointerException`, please address.